doc/reference/usermode/syscalls.rst

```diff
@@ -312,159 +287,32 @@ If any check fails, the macros will return a nonzero value. The macro
-the implementation function. If the implementation function returns a value,
-this needs to be returned by the handler, otherwise the handler should return
-0.
-
-.. note:: Do not forget that all the arguments to the handler are passed in as
-    unsigned 32-bit values.  If checks are needed on parameters that are
-    actually signed values, casts may be needed in order for these checks to
-    be performed properly.
-
-Using :c:func:`k_sem_init()` as an example again, we need to enforce that the
-semaphore object passed in is a valid semaphore object (but not necessarily
-initialized), and that the limit parameter is nonzero:
-
-.. code-block:: c
-
-    Z_SYSCALL_HANDLER(k_sem_init, sem, initial_count, limit)
-    {
-        Z_OOPS(Z_SYSCALL_OBJ_INIT(sem, K_OBJ_SEM));
-        Z_OOPS(Z_SYSCALL_VERIFY(limit != 0));
-        _impl_k_sem_init((struct k_sem *)sem, initial_count, limit);
-        return 0;
-    }
-
-Simple Handler Declarations
----------------------------
-
-Many kernel or driver APIs have very simple handler functions, where they
-either accept no arguments, or take one object which is a kernel object
-pointer of some specific type. Some special macros have been defined for
-these simple cases, with variants depending on whether the API has a return
-value:
-
-* :c:macro:`Z_SYSCALL_HANDLER1_SIMPLE()` one kernel object argument, returns
-  a value
-* :c:macro:`Z_SYSCALL_HANDLER1_SIMPLE_VOID()` one kernel object argument,
-  no return value
-* :c:macro:`Z_SYSCALL_HANDLER0_SIMPLE()` no arguments, returns a value
-* :c:macro:`Z_SYSCALL_HANDLER0_SIMPLE_VOID()` no arguments, no return value
-
-For example, :c:func:`k_sem_count_get()` takes a semaphore object as its
-only argument and returns a value, so its handler can be completely expressed
-as:
-
-.. code-block:: c
-
-    Z_SYSCALL_HANDLER1_SIMPLE(k_sem_count_get, K_OBJ_SEM, struct k_sem *);
-
-System Calls With 6 Or More Arguments
-=====================================
-
-System calls may have more than six arguments, however the number of arguments
-passed in via registers when the privilege elevation is invoked is fixed at six
-for all architectures. In this case, the sixth and subsequent arguments to the
-system call are placed into a struct, and a pointer to that struct is passed to
-the handler as its sixth argument.
-
-See ``include/syscall.h`` to see how this is done; the struct passed in must be
-validated like any other memory buffer. For example, for a system call
-with nine arguments, arguments 6 through 9 will be passed in via struct, which
-must be verified since memory pointers from user mode can be incorrect or
-malicious:
-
-.. code-block:: c
-
-    Z_SYSCALL_HANDLER(k_foo, arg1, arg2, arg3, arg4, arg5, more_args_ptr)
-    {
-        struct _syscall_9_args *margs = (struct _syscall_9_args *)more_args_ptr;
+All system calls are dispatched to a verifier function with a prefixed
+z_vrfy_ name based on the system call.  They have exactly the same
```

Comment:
in reST, a trailing underscore indicates an outgoing link, hence the doc build error saying there's no link target for the name ``z_vrfy``.  The way to fix this is to change this reference to:
```
... with a prefixed ``z_vrfy_`` name
```